tests/multi_ptr/multi_ptr_accessor_constructor.h

```diff
@@ -0,0 +1,147 @@
+    queue.submit([&](sycl::handler& cgh) {
+      auto same_type_acc =
+          same_type_buf.template get_access<sycl::access_mode::write>(cgh);
+      auto same_value_acc =
+          same_value_buf.template get_access<sycl::access_mode::write>(cgh);
+      auto accessor_instance = init_val_buffer.template get_access<Mode>(cgh);
+
+      cgh.single_task<kernel>([=] {
+        // Creating multi_ptr object with accessor constructor
+        multi_ptr_t mptr(accessor_instance);
+
+        auto id = util::get_cts_object::id<dimension>::get(0, 0, 0);
+        // Check that mptr points at same value as accessor
+        same_value_acc[0] = (*(mptr.get()) == accessor_instance[id]);
+
+        // Check that type of value is correct
+        same_type_acc[0] =
+            std::is_same_v<decltype(mptr.get()), typename multi_ptr_t::pointer>;
+      });
+    });
+  }
+  if (!same_type) {
+    FAIL(log, (get_case_description<dimension, Space, Mode,
+                                    sycl::access::target::device>(
+                  "Incorrect type", type_name)));
+  }
+  if (!same_value) {
+    FAIL(log, (get_case_description<dimension, Space, Mode,
+                                    sycl::access::target::device>(
+                  "Incorrect value", type_name)));
+  }
+}
+
+/** @brief Run tests for all combinations of access mode
+ */
+template <typename T, sycl::access::address_space Space, int dimension>
+void run_tests_for_dimension(sycl_cts::util::logger& log,
+                             const std::string& type_name) {
+  run_tests<T, Space, dimension, sycl::access::mode::read>(log, type_name);
+  run_tests<T, Space, dimension, sycl::access::mode::write>(log, type_name);
+  run_tests<T, Space, dimension, sycl::access::mode::read_write>(log,
+                                                                 type_name);
+}
+
+/** @brief Run tests for all combinations of dimensions
+ */
+template <typename T, sycl::access::address_space Space>
+void run_tests_for_space(sycl_cts::util::logger& log,
+                         const std::string& type_name) {
+  run_tests_for_dimension<T, Space, 1>(log, type_name);
+  run_tests_for_dimension<T, Space, 2>(log, type_name);
+  run_tests_for_dimension<T, Space, 3>(log, type_name);
+}
+
+/** @brief Provides verification of multi_ptr global_space and generic_space
+ * accessor constructor for specific type given
+ *  @tparam T Specific type of multi ptr data to use
+ */
+template <typename T>
+class check_multi_ptr_accessor_constructor_for_type {
+ public:
+  void operator()(sycl_cts::util::logger& log, const std::string& type_name) {
+    using namespace sycl::access;
+    // Accroding to spec of multi_ptr, constructor can take only
+    // global_space and generic_space accessors
+    run_tests_for_space<T, address_space::global_space>(log, type_name);
+    run_tests_for_space<T, address_space::generic_space>(log, type_name);
+  }
```

Comment:
The spec says these constructors only exist for certain address spaces so can we have a test which we expect to fail if a different address space is used?

Also are we missing the deprecated constructors?

Reply:
> test which we expect to fail if a different address space is used

Looks like a negative case to me. We usually do not cover negative cases in tests.

> Also are we missing the deprecated constructors?

We have different tests for legacy constructors [here](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/tests/multi_ptr/multi_ptr_legacy_constructors_common.h). Is this what you meant?

Reply:
> We usually do not cover negative cases in tests.

OK, thanks for confirming.

> We have different tests for legacy constructors here. Is this what you meant?

Yes, they look OK. Sorry I missed them.